engine/api/workflow/dao_run_secret.go

```diff
@@ -0,0 +1,43 @@
+package workflow
+
+import (
+	"context"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/sdk"
+)
+
+func InsertRunSecret(ctx context.Context, db gorp.SqlExecutor, wrSecret *sdk.WorkflowRunSecret) error {
+	dbData := &dbWorkflowRunSecret{WorkflowRunSecret: *wrSecret}
+	dbData.ID = sdk.UUID()
+	if err := gorpmapping.InsertAndSign(ctx, db, dbData); err != nil {
+		return err
+	}
+	*wrSecret = dbData.WorkflowRunSecret
+	return nil
+}
+
+func loadRunSecretByContext(ctx context.Context, db gorp.SqlExecutor, runID int64, context string) ([]sdk.Variable, error) {
```

Comment:
what is context ? i think "entity" is clearer

Reply:
removed